components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonInstallationDialogFragment.kt

```diff
@@ -0,0 +1,271 @@
+        positiveButton.setOnClickListener {
+            onPositiveButtonClicked?.invoke(addon, allowPrivateBrowsing)
+            dismiss()
+        }
+
+        if (positiveButtonBackgroundColor != DEFAULT_VALUE) {
+            val backgroundTintList =
+                    ContextCompat.getColorStateList(requireContext(), positiveButtonBackgroundColor)
+            positiveButton.backgroundTintList = backgroundTintList
+        }
+
+        if (positiveButtonTextColor != DEFAULT_VALUE) {
+            val color = ContextCompat.getColor(requireContext(), positiveButtonTextColor)
+            positiveButton.setTextColor(color)
+        }
+
+        if (positiveButtonRadius != DEFAULT_VALUE.toFloat()) {
+            val shape = GradientDrawable()
+            shape.shape = GradientDrawable.RECTANGLE
+            shape.setColor(
+                ContextCompat.getColor(
+                    requireContext(),
+                    positiveButtonBackgroundColor
+                )
+            )
+            shape.cornerRadius = positiveButtonRadius
+            positiveButton.background = shape
+        }
+
+        return rootView
+    }
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun fetchIcon(addon: Addon, iconView: ImageView, scope: CoroutineScope = this.scope): Job {
+        return scope.launch {
+            try {
+                val iconBitmap = addonCollectionProvider.getAddonIconBitmap(addon)
+                iconBitmap?.let {
+                    scope.launch(Dispatchers.Main) {
+                        iconView.setImageDrawable(BitmapDrawable(iconView.resources, it))
+                    }
+                }
+            } catch (e: IOException) {
+                scope.launch(Dispatchers.Main) {
+                    val context = iconView.context
+                    val att = context.theme.resolveAttribute(android.R.attr.textColorPrimary)
+                    iconView.setColorFilter(ContextCompat.getColor(context, att))
+                    iconView.setImageDrawable(context.getDrawable(R.drawable.mozac_ic_extensions))
+                }
+                logger.error("Attempt to fetch the ${addon.id} icon failed", e)
+            }
+        }
+    }
+
+    @Suppress("LongParameterList")
+    companion object {
+        /**
+         * Returns a new instance of [AddonInstallationDialogFragment].
+         * @param addon The addon to show in the dialog.
+         * @param promptsStyling Styling properties for the dialog.
+         * @param onPositiveButtonClicked A lambda called when the allow button is clicked.
+         */
+        fun newInstance(
+            addon: Addon,
+            addonCollectionProvider: AddonCollectionProvider,
+            promptsStyling: PromptsStyling? = PromptsStyling(
+                gravity = Gravity.BOTTOM,
+                shouldWidthMatchParent = true
+            ),
+            onPositiveButtonClicked: ((Addon, Boolean) -> Unit)? = null
```

Comment:
nit: see above re: naming. This lambda gets invoked both the confirm button is clicked and also when the dialog is dismissed.